pkg/controller/release.go

```diff
@@ -237,14 +245,22 @@ func (e *helmExternal) Update(ctx context.Context, mg resource.Managed) (managed
 		return managed.ExternalUpdate{}, errors.Wrap(err, errFailedToLoadPatches)
 	}
 
+	updateToLatest := cr.Spec.ForProvider.Chart.Version == ""
 	rel, err := e.helm.Upgrade(meta.GetExternalName(cr), cd, cv, p)
 	if err != nil {
 		return managed.ExternalUpdate{}, errors.Wrap(err, errFailedToUpgrade)
 	}
+
 	if rel == nil {
 		return managed.ExternalUpdate{}, errors.New(errLastReleaseIsNil)
 	}
 
+	if updateToLatest {
```

Comment:
Instead of passing chart spec as reference and updating version implicitly, or, returning version as yet another parameter explicitly, we could just get the version from release at this point as:

```
cr.Spec.ForProvider.Chart.Version = rel.Chart.Metadata.Version
```  

Then, we even wouldn't need to infer version from filename etc.

One possible problem with this would be only to be able to get/set version after a successful `helm install/upgrade`, but it looks like this is already the case and solution for that seems like to separate `pullAndLoad` as yet another function in helm client interface instead of embedding into `Install` and `Upgrade` functions. Then we could get version from chart metadata instead of release metadata as `chart.Metadata.Version` after calling `PullAndLoad` before Install/Upgrade.

Reply:
Good point! Updated the interface and use the version from the chart.